### PR TITLE
DEV-120566 Escalation Chain

### PR DESCRIPTION
**Pull request by lm-madhvi**
Created a functionality to add , update and remove escalation chains to LogicMonitor through the LogicMonitor Provider in Terraform.

### CODE REVIEW

File: logicmonitor/provider.go [hunk 1]
```diff
@@ -31,16 +31,17 @@ func Provider() *schema.Provider {
 				Type:        schema.TypeBool,
 				Optional:    true,
 				Default:     false,
-				Description: "(Experimental) true if going for bulk resource, default is false",
```
Comment (lmswagatadutta) [inline logicmonitor/provider.go]: Let's keep it as `Experimental`

File: logicmonitor/schemata/cloud_services_schema.go [hunk 2]
```diff
@@ -15,806 +14,807 @@ func CloudServicesSchema() map[string]*schema.Schema {
 			},
 			Optional: true,
 		},
-
+
```
Comment (lmswagatadutta) [inline logicmonitor/schemata/cloud_services_schema.go]: It looks like this file only has whitespace changes, which are not required; please revert the changes for this file

File: logicmonitor/schemata/widget_token_schema.go [hunk 3]
```diff
@@ -2,35 +2,35 @@ package schemata
```
Comment (lmswagatadutta) [inline logicmonitor/schemata/widget_token_schema.go]: This file also has whitespace changes.

File: models/dashboard.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ import (
 )
 
 // Dashboard dashboard
+// Example: isResource
```
Comment (lmswagatadutta) [inline models/dashboard.go]: Please add only those files related to the escalation chain.

File: models/dashboard_group.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ import (
 )
 
 // DashboardGroup dashboard group
+// Example: isResource
```
Comment (lmswagatadutta) [inline models/dashboard_group.go]: Please add only those files related to the escalation chain.

File: models/device_group.go [hunk 6]
```diff
@@ -16,6 +16,7 @@ import (
 )
 
 // DeviceGroup device group
+// Example: isResource
```
Comment (lmswagatadutta) [inline models/device_group.go]: Please add only those files related to the escalation chain.

**Review by lmswagatadutta — CHANGES_REQUESTED** [review]
@lm-madhvi I've gone through the changes you made, and overall, it looks quite promising. You've done a great job in implementing the new feature and addressing some of the initial concerns raised during the initial review. However, there are a few areas that still require some attention and improvements before we can merge this into the main branch.

File: logicmonitor/provider.go (hunk 1 above)
Comment (lm-madhvi) [reply logicmonitor/provider.go]: Done

File: models/device_group.go (hunk 6 above)
Comment (lm-madhvi) [reply models/device_group.go]: Done

File: models/dashboard_group.go (hunk 5 above)
Comment (lm-madhvi) [reply models/dashboard_group.go]: Done

File: models/dashboard.go (hunk 4 above)
Comment (lm-madhvi) [reply models/dashboard.go]: Done

File: logicmonitor/schemata/widget_token_schema.go (hunk 3 above)
Comment (lm-madhvi) [reply logicmonitor/schemata/widget_token_schema.go]: Removed whitespaces

**Comment by lm-madhvi** [comment]
> @lm-madhvi I've gone through the changes you made, and overall, it looks quite promising. You've done a great job in implementing the new feature and addressing some of the initial concerns raised during the initial review. However, there are a few areas that still require some attention and improvements before we can merge this into the main branch.

@lmswagatadutta Thanks for all the support and guidance. I worked on the changes required and resolved them. Please check the latest commit .

**Review by lmswagatadutta — APPROVED** [review]
